### ISSUE TITLE
Parametrized Unicode Strings Don't Work in Python2.7

### ISSUE THREAD

**Issue by KenStorey** [taverntesting/tavern#269]
The following test works as expected :-
```
# test_bug.tavern.yaml
---

test_name: Bug Demonstration

marks:
  - bugdemo

paho-mqtt:
  client:
    client_id: tavern-tester
  connect:
    host: localhost
    port: 1883
    timeout: 3

stages:

  - name: Step 1 - Display Line

    mqtt_publish:
      topic: dcp/target_display_line
      json:
        line_no: !int 1
        line_text: "手机号格式不正确............"

    mqtt_response:
      topic: dcp/dcp_response
      json:
        result_code: !int 0
      timeout: 5
```
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.6, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /media/sf_GIT_Repository/TDCPSimulator/Tests, inifile:
plugins: tavern-0.22.1
collected 62 items / 61 deselected                                             

test_bug.tavern.yaml::Bug Demonstration PASSED                           [100%]

=================== 1 passed, 61 deselected in 1.12 seconds ====================

```

However, if _line_text_ is parametrized it fails with a 'UnicodeEncodeError' :-
```
# test_bug.tavern.yaml
---

test_name: Bug Demonstration

marks:
  - bugdemo
  - parametrize:
      key: [line, text]
      vals:
        - [1, "手机号格式不正确............"]
        - [2, "Happy Chinese New Year"]

paho-mqtt:
  client:
    client_id: tavern-tester
  connect:
    host: localhost
    port: 1883
    timeout: 3

stages:

  - name: Step 1 - Display Line

    mqtt_publish:
      topic: dcp/target_display_line
      json:
        line_no: !int "{line}"
        line_text: "{text}"

    mqtt_response:
      topic: dcp/dcp_response
      json:
        result_code: !int 0
      timeout: 5

```
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.6, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /media/sf_GIT_Repository/TDCPSimulator/Tests, inifile:
plugins: tavern-0.22.1
collected 61 items / 1 errors / 61 deselected                                  

==================================== ERRORS ====================================
____________________ ERROR collecting test_bug.tavern.yaml _____________________
/usr/local/lib/python2.7/dist-packages/tavern/testutils/pytesthook.py:400: in collect
    for i in self._generate_items(test_spec):
/usr/local/lib/python2.7/dist-packages/tavern/testutils/pytesthook.py:368: in _generate_items
    test_spec, parametrize_marks, pytest_marks
/usr/local/lib/python2.7/dist-packages/tavern/testutils/pytesthook.py:254: in get_parametrized_items
    inner_formatted = inner_fmt.format(*flattened_values)
E   UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-7: ordinal not in range(128)
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
==================== 61 deselected, 1 error in 0.51 seconds ====================
```

**Comment by michaelboulton** [comment]
This should now be fixed in version 0.24.0.

**Comment by KenStorey** [comment]
This seems to have fixed the reported problem in Tavern but, unfortunately, in 2.7 the issue just gets kicked further down the road:

```
  File "/usr/lib/python2.7/json/encoder.py", line 361, in _iterencode_dict
    items = sorted(dct.items(), key=lambda kv: kv[0])
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 51: ordinal not in range(128)
```

Not a Tavern problem though.
